src/main/java/frc/robot/commands/intake/RunIntake.java

```diff
@@ -1,46 +1,39 @@
 package frc.robot.commands.intake;
 
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 import edu.wpi.first.wpilibj2.command.Command;
 import frc.robot.constants.RobotConfig;
+import frc.robot.subsystems.indexer.Indexer;
 import frc.robot.subsystems.intake.Intake;
 
 public class RunIntake extends Command {
   private final Intake m_intake;
+  private final Indexer m_indexer;
   private boolean m_reversed;
 
-  /**
-   * Creates a new RunIntake command, which runs the roller motor on the intake subsystem to intake
-   * a note
-   *
-   * @param intake The subsystem used by this command.
-   */
-  public RunIntake(Intake intake, boolean reversed) {
+  public RunIntake(Intake intake, Indexer indexer, boolean reversed) {
     m_intake = intake;
+    m_indexer = indexer;
     m_reversed = reversed;
 
     addRequirements(intake);
   }
 
-  // Called when the command is initially scheduled.
   @Override
   public void initialize() {
     int multiplier = m_reversed ? -1 : 1;
     m_intake.run(RobotConfig.IntakeConfig.kDefaultSpeed * multiplier);
+    m_indexer.startFeedNote(m_reversed);
   }
 
-  // Called every time the scheduler runs while the command is scheduled.
-  @Override
-  public void execute() {}
-
-  // Called once the command ends or is interrupted.
   @Override
   public void end(boolean interrupted) {
     m_intake.stop();
+    m_indexer.stopFeedNote();
   }
 
-  // Returns true when the command should end.
   @Override
   public boolean isFinished() {
-    return false;
+    return SmartDashboard.getBoolean("Shooter/line break", false);
```

Comment:
This looks like a bad coding convention. I recommend you create a method in the Shooter subsystem that returns the current state of the line break sensor and then use that value to determine the sensor's state.

Reply:
Additionally, if you move the line break sensor to the indexer subsystem, you would not have to add any extra command requirements and use the already available `m_indexer` value.